### PR TITLE
qn - *not* filtering support

### PR DESCRIPTION
**Pull request by ondratra**
This PR brings support for negative filtering in QN queries + fixed variant `or` filtering  https://github.com/Joystream/hydra/issues/485 & https://github.com/Joystream/hydra/issues/476.

This type of queries can be used now
```
# selects all nfts except the one with id 3
query {
  ownedNfts(
     where: {
          NOT: [
            {id_eq: 3},
          ]
        }
    ) {
      id,
      transactionalStatus {
        __typename
      }
    }
  }
}
```

Includes work from PRs #3815 and #3819.

### CODE REVIEW

**Comment by ondratra** [comment]
CI tests are failing on some connection issues that shouldn't be related to the PR itself. I'll try to do an extra commit later on to see if problems disappear. So far, it failed twice, with ~10 hours between runs.

**Review by zeeshanakram3 — COMMENTED** [review]
Not able to build `query-node` locally. Running 
- `yarn build` in `/joystream/query-node`, OR
- `yarn workspace query-node-root build` in `/joystream/`

results in the following error:
```
3/6] Fetching packages...
info No lockfile found.
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/5] Validating package.json...
[2/5] Resolving packages...
warning codecov@3.8.3: https://about.codecov.io/blog/codecov-uploader-deprecation-plan/
warning i18n-tag-schema > babel-runtime > core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
[3/5] Fetching packages...
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote --tags --heads git://github.com/axa-ch/cyclic-schemas-test.git
Directory: /workspace/joystream/query-node
Output:
fatal: unable to connect to github.com:
github.com[0: 140.82.121.4]: errno=Connection timed out
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Command failed with exit code 128.
```

File: cli/src/Api.ts [hunk 1]
```diff
@@ -469,10 +469,14 @@ export default class Api {
...
-      value,
-    ])
+    return (await this._api.query.storage.dataObjectsById.entries(bagId)).map(
+      ([
+        {
+          args: [, dataObjectId],
+        },
+        value,
+      ]) => [dataObjectId, value]
+    )
```
Comment (zeeshanakram3) [inline cli/src/Api.ts:479]: This formatting change causes the prettier issue
image

File: cli/src/Api.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ export default class Api {
 
   // Get api for use-cases where no type augmentations are desirable
   public getUnaugmentedApi(): UnaugmentedApiPromise {
-    return (this._api as unknown) as UnaugmentedApiPromise
+    return this._api as unknown as UnaugmentedApiPromise
```
Comment (zeeshanakram3) [inline cli/src/Api.ts:75]: This formatting change causes the prettier issue
image

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]

**Review by zeeshanakram3 — APPROVED** [review]

File: cli/src/Api.ts (hunk 1 above)
Comment (ondratra) [reply cli/src/Api.ts:479]: After a discussion, we've concluded that this issue was due to not properly updated `yarn` dependencies on local machine (removing all `node_modules` folders and running `yarn` again fixed the issue).

File: cli/src/Api.ts (hunk 2 above)
Comment (ondratra) [reply cli/src/Api.ts:75]: After a discussion, we've concluded that this issue was due to not properly updated `yarn` dependencies on local machine (removing all `node_modules` folders and running `yarn` again fixed the issue).

**Comment by ondratra** [comment]
The last commit had to fix an issue with TypeORM that (likely) changed the behavior for `find*` function using `logical or` in `where` condition. 

For example, the following `.get` call
```
const channel = await store.get(Channel, {
  where: channelAssets.map((assetName) => ({
    [assetName]: {
      id: dataObject.id,
  })),
  relations: [...channelAssets],
})
```
resulted in sql query
```
SELECT 
"Channel"."id" AS "Channel_id", "Channel__Channel_avatarPhoto"."id", "Channel__Channel_coverPhoto"."id"
...
FROM "channel" "Channel" 
INNER JOIN "storage_data_object" "Channel__Channel_avatarPhoto" ON "Channel__Channel_avatarPhoto"."id"="Channel"."avatar_photo_id"  
INNER JOIN "storage_data_object" "Channel__Channel_coverPhoto" ON "Channel__Channel_coverPhoto"."id"="Channel"."cover_photo_id" 
WHERE (("Channel__Channel_coverPhoto"."id" = '12641') OR ("Channel__Channel_avatarPhoto"."id" = '12641'))
```
but those two `INNER JOIN`s can't work with logical `OR` in the `WHERE` condition because they ultimately act as logical `AND`.

**Pull request merged**